libs/remix-core-plugin/src/lib/compiler-artefacts.ts

```diff
@@ -72,6 +72,53 @@ export class CompilerArtefacts extends Plugin {
     return contractsData
   }
 
+  async getArtefactsFromFE (path, contractName) {
+    const dirList = await this.call('fileManager', 'dirList', path)
+    if(dirList && dirList.length) {
+      if(dirList.includes(path + '/artifacts')) {
+        const fileList = await this.call('fileManager', 'fileList', path + '/artifacts')
+        const artefactsFilePaths = fileList.filter(filePath => {
+          const filenameArr = filePath.split('/')
+          const filename = filenameArr[filenameArr.length - 1]
+          if (filename === `${contractName}.json` || filename === `${contractName}_metadata.json`) return true
+        })
+        if (artefactsFilePaths && artefactsFilePaths.length) {
+          const content = await this.call('fileManager', 'readFile', artefactsFilePaths[1])
+          const artifacts = JSON.parse(content)
+          return { abi: artifacts.abi, bytecode: artifacts.data.bytecode.object }
```

Comment:
what's the reason why you can't just return the compilation result? is this that the json files from the file system don't exactly match with the raw compilation result?

Reply:
in that case, we should maybe think of replacing the <contract>_metadata.json by the actual compilation result

Reply:
The file named as <contractName>.json does not store bytecode. We can think of putting the compilation result in a file that can be done in a different PR

Reply:
Implemented by https://github.com/ethereum/remix-project/pull/2081